java/client/src/main/java/glide/api/models/commands/SetOptions.java

```diff
@@ -49,11 +53,51 @@ public enum ConditionalSet {
          * Only set the key if it does not already exist. Equivalent to <code>NX</code> in the Valkey
          * API.
          */
-        ONLY_IF_DOES_NOT_EXIST("NX");
+        ONLY_IF_DOES_NOT_EXIST("NX"),
+        /**
+         * Only set the key if the current value of key equals the {@link SetOptions#comparisonValue}.
+         * Equivalent to <code>IFEQ comparison-value</code> in the Valkey API.
+         */
+        ONLY_IF_EQUAL("IFEQ");
 
         private final String valkeyApi;
     }
 
+    // Builder class for SetOptions
+    public static class SetOptionsBuilder {
+        private ConditionalSet conditionalSet;
+        private String comparisonValue;
+
+        /**
+         * Set the condition for the value to be set
+         *
+         * @param conditionalSet the condition to set (ONLY_IF_EXISTS, ONLY_IF_DOES_NOT_EXIST)
+         * @return this builder instance
+         * @throws IllegalArgumentException if the conditionalSet is ONLY_IF_EQUAL
+         */
+        public SetOptionsBuilder conditionalSet(ConditionalSet conditionalSet) {
+            if (conditionalSet == ConditionalSet.ONLY_IF_EQUAL) {
+                throw new IllegalArgumentException(
+                        "For ONLY_IF_EQUAL, use the conditionalSetIfEqualTo(String value) method.");
+            }
+            this.conditionalSet = conditionalSet;
+            this.comparisonValue = null; // Clear comparisonValue when not using ONLY_IF_EQUAL
+            return this;
+        }
+
+        /**
+         * Set the condition to ONLY_IF_EQUAL and specify the comparison value
+         *
+         * @param value the value to compare
+         * @return this builder instance
```

Comment:
```suggestion
         * Set the key if the comparison value matches the existing value.
         *
         * @since Valkey 8.1 and above.
         * @param value The value to compare.
         * @return This builder instance.
```